pkg/hooks/loader.go

```diff
@@ -533,14 +533,18 @@ func (h *Hook) StopUserApplication() {
 	h.SetUserAppTerminateInitiated(true)
 	if h.userAppCmd != nil && h.userAppCmd.Process != nil {
 		h.logger.Debug("the process state for the user process", zap.String("state", h.userAppCmd.ProcessState.String()), zap.Any("processState", h.userAppCmd.ProcessState))
-		if h.userAppCmd.ProcessState != nil && h.userAppCmd.ProcessState.Exited() {
+
+		// passing userAppCmd without /usr/bin/sh -c
+		dockerCmdBool, dockerCmd := utils.IsDockerRelatedCmd(h.userAppCmd.String()[15:])
+		if h.userAppCmd.ProcessState != nil && h.userAppCmd.ProcessState.Exited() && !dockerCmdBool {
```

Comment:
Just wanted to clarify that here we are checking if command is docker related or not, since in docker start as well in docker detached mode the process is exited but still we have to remove the container.